### PR TITLE
Modify PasswordManager to allow BasicTextEncryptor

### PR DESCRIPTION
**Pull request by zliu41**
Also modified the encryption tool to allow reading master password from file.

### CODE REVIEW

File: gobblin-api/src/main/java/gobblin/password/PasswordManager.java [hunk 1]
```diff
@@ -63,12 +67,17 @@ public PasswordManager load(Optional<String> masterPassword) {
             }
           });
 
-  private final Optional<StrongTextEncryptor> encryptor;
+  private Optional<TextEncryptor> encryptor;
 
   private PasswordManager(Optional<String> masterPassword) {
```
Comment (chavdar) [inline gobblin-api/src/main/java/gobblin/password/PasswordManager.java]: Please pass the encryptor class as an optional argument in the constructor. Using a static field creates all kinds of race conditions.

**Comment by zliu41** [comment]
@chavdar updated. Changed key of instance cache from `Optional<String>` to `Map.Entry<Optional<String>, Boolean>`, where true means strong encryptor and false means basic encryptor.

**Comment by chavdar** [comment]
Please add a unit test to check the new plain/strong encryption functionality.

**Comment by zliu41** [comment]
@chavdar updated.

**Comment by chavdar** [comment]
LGTM.

**Pull request merged**